components/lib/crash/src/test/java/mozilla/components/lib/crash/service/GleanServiceTest.kt

```diff
@@ -0,0 +1,63 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.lib.crash.service
+
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import androidx.work.testing.WorkManagerTestInitHelper
+import mozilla.components.lib.crash.Crash
+import mozilla.components.lib.crash.GleanMetrics.CrashMetrics
+import mozilla.components.service.glean.Glean
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import org.junit.Before
+import org.junit.runner.RunWith
+import org.mockito.Mockito.spy
+import org.mockito.Mockito.verify
+import org.mockito.Mockito.verifyNoMoreInteractions
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class GleanServiceTest {
```

Comment:
I'd add a test to simulate a crash/recording before Glean is initialized, to make sure this doesn't act weird.

Reply:
It doesn't act weird or crash or anything, but it doesn't record anything either.  Basically, it looks like we won't be able to record any crashes that occur before Glean is initialized without thinking around the problem a bit.

Reply:
We don't need to record before Glean is initialized I think, but we don't want to crash either.

Reply:
Intuitively, it feels like for MVP it's good enough to only record crashes the occur after `Glean.initialize`...

Reply:
So a simple test to show that lib-crash attempting to record before Glean is initialized doesn't crash and records no data, reasonable enough.

Reply:
That way if we change it to where we can record data before Glean is initialized, we can just modify the test a bit to show the collection.